File: server/src/main/java/pl/ordovita/surveys/domain/exception/SurveyException.java
package pl.ordovita.surveys.domain.exception;

public class SurveyException extends RuntimeException{
    public SurveyException(String message) {
        super(message);
    }
    public SurveyException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: DotNetServer/Ordovita.Domain/Surveys/Exception/SurveyException.cs
using Ordovita.Domain.Common;

namespace Ordovita.Domain.Surveys.Exception;

public class SurveyException
{
    public static readonly Error MissingTitle =
        Error.Validation("Survey.MissingTitle", "Title is required.");

    public static readonly Error MissingDescription =
        Error.Conflict("Survey.MissingDescription", "Description is required.");

    public static readonly Error NotFound =
        Error.NotFound("Survey.NotFound", "Survey was not found.");

    public static readonly Error AlreadyVisable =
        Error.NotFound("Survey.AlreadyVisable", "Survey is already visible.");

    public static readonly Error AlreadyNotVisable =
        Error.NotFound("Survey.AlreadyNotVisable", "Survey is not already visible.");
}